packages/api/src/hooks/UseCennznet.ts

```diff
@@ -0,0 +1,128 @@
+
+import { web3Accounts, web3Enable } from '@polkadot/extension-dapp';
+import { InjectedAccountWithMeta, MetadataDef } from '@polkadot/extension-inject/types';
+import { getSpecTypes } from '@polkadot/types-known';
+import { defaults as addressDefaults } from '@polkadot/util-crypto/address/defaults';
+import { default as cennznetExtensions } from '@cennznet/types/interfaces/extrinsic/signedExtensions/cennznet';
+import { ApiRx } from '@cennznet/api';
+
+import { ApiOptions } from '../types';
+import { Api, Api as ApiPromise } from '../Api';
+
+export async function UseCennznetRx(
+  dAppName: string,
+  options: ApiOptions
+): Promise<{ api: ApiRx; accounts: InjectedAccountWithMeta[]; isExtensionInstalled: boolean }> {
+  const api = await ApiRx.create(options).toPromise();
+  return updateExtensionMetadataRetrieveAccounts(api, options, dAppName);
+}
+
+export async function UseCennznet(
+  dAppName: string,
+  options: ApiOptions
+): Promise<{ api: ApiPromise; accounts: InjectedAccountWithMeta[]; isExtensionInstalled: boolean }> {
+  const api = await Api.create(options);
+  return updateExtensionMetadataRetrieveAccounts(api, options, dAppName);
+}
+
+async function updateExtensionMetadataRetrieveAccounts(
+  api: ApiPromise,
+  options: ApiOptions,
+  dAppName: string
+): Promise<{ api: ApiPromise; accounts: InjectedAccountWithMeta[]; isExtensionInstalled: boolean }>;
+
+async function updateExtensionMetadataRetrieveAccounts(
+  api: ApiRx,
+  options: ApiOptions,
+  dAppName: string
+): Promise<{ api: ApiRx; accounts: InjectedAccountWithMeta[]; isExtensionInstalled: boolean }>;
+
+async function updateExtensionMetadataRetrieveAccounts(
+  api: ApiPromise | ApiRx,
+  options: ApiOptions,
+  dAppName: string
+): Promise<{ api: ApiPromise | ApiRx; accounts: InjectedAccountWithMeta[]; isExtensionInstalled: boolean }> {
+  const extensions = await web3Enable(dAppName);
+  if (extensions.length === 0) {
+    return { api: api, accounts: null, isExtensionInstalled: false };
+  }
+  const polkadotExtension = extensions.find((ext) => ext.name === 'polkadot-js');
+  const metadata = polkadotExtension.metadata;
+
+  const checkIfMetaUpdated = localStorage.getItem(
+    `cennznet-ext-meta-${options.network}-${api.runtimeVersion.specName}-${api.runtimeVersion.specVersion}`
+  );
+  if (!checkIfMetaUpdated) {
+    const metadataDef = await extractMeta(api);
+    await metadata.provide(metadataDef);
+    localStorage.setItem(
+      `cennznet-ext-meta-${options.network}-${api.runtimeVersion.specName}-${api.runtimeVersion.specVersion}`,
+      'true'
+    );
+  }
+  const allAccounts = await web3Accounts();
+  return { api: api, accounts: allAccounts, isExtensionInstalled: true };
+}
+
+async function extractMeta(api) {
+  const systemChain = await api.rpc.system.chain();
+  const specTypes = getSpecTypes(
+    api.registry,
+    systemChain,
+    api.runtimeVersion.specName,
+    api.runtimeVersion.specVersion
+  );
+  if (specTypes.ExtrinsicSignatureV4) {
```

Comment:
can remove this delete specific class template with 
```

// remove all the classes from the spectypes
// the metadata that gets updated on the cennznet extension requires types of type (Record<string, string|object>)
                        const filteredSpecTypes = Object.keys(specTypes)
                            .filter(key => {
                                if (typeof specTypes[key] === 'function') {
                                    return false;
                                }
                                return true;
                            })
                            .reduce((obj, key) => {
                                obj[key] = specTypes[key];
                                return obj;
                            }, {});
```

Reply:
which would mean, in future even if we add new classes we won't need to add a delete that particular class line here..
The code will still work...

Reply:
oh yup cool just updated